tests/testthat/test_dotproduct.R

```diff
@@ -0,0 +1,48 @@
+test_that("dotproduct", {
+    
+    ## create two dummy spectra (aligned)
+    x <- data.frame(mz=c(100.002, 100.001, NA, 300.01, 300.02, NA), 
+             intensity=c(2, 1.5, 0, 1.2, 0.9, 0))
+    y <- data.frame(mz=c(100.0, NA, 200.0, 300.002, 300.025, 300.0255),
+             intensity=c(2, 0, 3, 1, 4, 0.4))
+    
+    ## calculate similarity values
+    expect_warning(dotproduct(x, y, m=0, n=2))
```

Comment:
Please at a few words as regular expression to your `expect_warning` and `expect_error` test, e.g.:
```suggestion
    expect_warning(dotproduct(x, y, m=0, n=2), "not identical")
```

Reply:
added regular expression for `expect_warning`and `expect_error` tests